gst/tensor_filter/tensor_filter_tensorflow_core.cc

```diff
@@ -58,70 +55,255 @@ TFCore::~TFCore ()
+ *        -3 if the type of input tensors is not matched.
+ *        -4 if the dimension of input tensors is not matched.
  */
 int
-TFCore::getOutputTensorDim (GstTensorsInfo * info)
+TFCore::inputTensorValidation (std::vector<const NodeDef*> placeholders)
 {
-  /**
-   * @todo fill here
-   */
+  if (inputTensorMeta.num_tensors != placeholders.size()){
+    GST_ERROR ("Input Tensor is not valid: the number of input tensor is different\n");
+    return -1;
+  }
+  int length = placeholders.size();
+  for (int i = 0; i < length; i++) {
+    const NodeDef* node = placeholders[i];
+    string shape_description = "None";
+    if (node->attr().count("shape")) {
+      TensorShapeProto shape_proto = node->attr().at("shape").shape();
+      Status shape_status = PartialTensorShape::IsValidShape(shape_proto);
+      if (shape_status.ok()) {
+        shape_description = PartialTensorShape(shape_proto).DebugString();
+      } else {
+        shape_description = shape_status.error_message();
+      }
+    }
+    char chars[] = "[]";
+    for (unsigned int i = 0; i < strlen(chars); ++i)
+    {
+      shape_description.erase (
+        std::remove(
+          shape_description.begin(),
+          shape_description.end(),
+          chars[i]
+        ),
+        shape_description.end()
+      );
+    }
+
+    DataType dtype = DT_INVALID;
+    if (node->attr().count("dtype")) {
+      dtype = node->attr().at("dtype").type();
+    }
+
+    if (strcmp (inputTensorMeta.info[i].name, node->name().c_str())){
+      GST_ERROR ("Input Tensor is not valid: the name of input tensor is different\n");
+      return -2;
+    }
+    if (inputTensorMeta.info[i].type != getTensorTypeFromTF(dtype)){
+      GST_ERROR ("Input Tensor is not valid: the type of input tensor is different\n");
+      return -3;
+    }
+
+    gchar **str_dims;
+    str_dims = g_strsplit (shape_description.c_str(), ",", -1);
+    for(int j = 0; j < NNS_TENSOR_RANK_LIMIT; j++){
+      if (!strcmp (str_dims[j], "?"))
+        continue;
+      
+      if (inputTensorMeta.info[i].dimension[NNS_TENSOR_RANK_LIMIT - j - 1] != atoi (str_dims[j])){
+        GST_ERROR ("Input Tensor is not valid: the dim of input tensor is different\n");
+        return -4;
+      }
+    }
+  }
   return 0;
 }
 
 /**
- * @brief	return the Dimension of Tensor.
- * @param tensor_idx	: the real index of model of the tensor
- * @param[out] dim	: the array of the tensor
- * @param[out] type	: the data type of the tensor
+ * @brief extract and store the information of src tensors
  * @return 0 if OK. non-zero if error.
  */
 int
-TFCore::getTensorDim (tensor_dim dim, tensor_type * type)
+TFCore::setTensorProp (GstTensorsInfo * dest, GstTensorsInfo src)
 {
-
+  dest->num_tensors = src.num_tensors;
+  for(int i = 0; i < src.num_tensors; i++){
+    dest->info[i].name = src.info[i].name;
+    dest->info[i].type = src.info[i].type;
+    dest->info[i].dimension[0] = src.info[i].dimension[0];
```

Comment:
add loop to remove the dependency of rank limit

Reply:
it's applied